src/compile/mark-bar.ts

```diff
@@ -159,7 +159,33 @@ export namespace bar {
   }
 
   export function labels(model: Model) {
-    // TODO(#64): fill this method
-    return undefined;
+    let p: any = {};
+
+    const orient = model.config().mark.orient;
+    const isHorizontal = (orient === 'horizontal');
+
+    // for reactive geometry
+    let datumField = 'datum.' + (isHorizontal ? model.field(X) : model.field(Y));
```

Comment:
you can also do

`const datumField = model.field(isHorizontal ? X :Y, {datum: 1});`

- Should be `const` because you never modify it.  
- See `FieldRefOption` object which is an optional parameter for `model.field`.

- To make it work correct with stacked bar, need to add suffix `’_end’`.   
See `mark-bar.ts` line 27.   
  - (We should make `suffix` another parameter in `FieldRefOption` though! — feel free to do that in a separate PR — since that refactor should be merged to the master)